pyroute2/dhcp/hooks.py

```diff
@@ -24,21 +25,54 @@ async def unbound(self, lease: Lease):
 
 class ConfigureIP(Hook):
     async def bound(self, lease: Lease):
-        LOG.info('STUB: add %s to %s', lease.ip, lease.interface)
-        # await ip(
-        #     "addr",
-        #     "replace",
-        #     f"{lease.ip}/{lease.subnet_mask}",
-        #     "dev",
-        #     lease.interface,
-        # )
+        LOG.debug('Adding IP %s to %s', lease.ip, lease.interface)
+        async with AsyncIPRoute() as ipr:
+            await ipr.addr(
+                "replace",
+                index=await ipr.link_lookup(ifname=lease.interface),
+                address=lease.ip,
+                mask=lease.subnet_mask,
+                broadcast=lease.broadcast_address
+            )
 
     async def unbound(self, lease: Lease):
-        LOG.info('STUB: remove %s from %s', lease.ip, lease.interface)
-        # await ip(
-        #     "addr",
-        #     "del",
-        #     f"{lease.ip}/{lease.subnet_mask}",
-        #     "dev",
-        #     lease.interface,
-        # )
+        LOG.debug('Removing IP %s from %s', lease.ip, lease.interface)
+        async with AsyncIPRoute() as ipr:
+            await ipr.addr(
+                "del",
+                index=await ipr.link_lookup(ifname=lease.interface),
+                address=lease.ip,
+                mask=lease.subnet_mask,
+                broadcast=lease.broadcast_address
+            )
+
+
+class ConfigureDefaultRoute(Hook):
+    async def bound(self, lease: Lease):
+        if lease.default_gateway is None:
+            LOG.error("Lease doesn't contain default gateway")
+            return
+        LOG.debug('Adding %s as default route through %s',
+                  lease.default_gateway, lease.interface)
+        async with AsyncIPRoute() as ipr:
+            ifindex = await ipr.link_lookup(ifname=lease.interface),
+            await ipr.route(
+                "replace",
+                dst="0.0.0.0/0",
+                gateway=lease.default_gateway,
+                oif=ifindex
+            )
+
+    async def unbound(self, lease: Lease):
+        if lease.default_gateway is None:
+            LOG.error("Lease doesn't contain default gateway")
+            return
+        LOG.debug('Removing %s as default route.', lease.default_gateway)
+        async with AsyncIPRoute() as ipr:
+            ifindex = await ipr.link_lookup(ifname=lease.interface)
+            await ipr.route(
```

Comment:
Can it possible that the route will be removed automatically before this hook gets called? By the kernel or by some 3rd party software like NetworkManager.

Because then `ip.route('del', …)` will raise a `NetlinkError(code=errno.ENOENT)` here.

Reply:
Yes, a try/except with a warning log would be great here.